docs/user_guide.md

```diff
@@ -232,81 +451,214 @@ CodeChecker will get the hardcoded values for the compilers set in the
 export CC_LOGGER_GCC_LIKE="gcc:g++:clang"
 ~~~~~~~~~~~~~~~~~~~~~
 
-If there are still compilaton errors after using the `--add-compiler-defaults` flag
-it is possible that the wrong build target architecture (32bit, 64bit) is used. Please try to forward these compilation flags to the analyzers:
-- `-m32` (32bit build)
-- `-m64` (64bit build)
+If there are still compilation errors after using the `--add-compiler-defaults`
+argument, it is possible that the wrong build target architecture
+(32bit, 64bit) is used. Please try to forward these compilation flags
+to the analyzers:
 
-See the Forward compiler options section how to do this.
+ - `-m32` (32-bit build)
+ - `-m64` (64-bit build)
 
-#### Forward compiler options
+#### Forwarding compiler options
 
-These options can modify the compilation actions logged by the build logger or
-created by cmake (exporting compile commands). The extra compiler options can be
-given in config files which are provided by the flags described below.
+Forwarded options can modify the compilation actions logged by the build logger
+or created by CMake (when exporting compile commands). The extra compiler
+options can be given in config files which are provided by the flags
+described below.
 
 The config files can contain placeholders in `$(ENV_VAR)` format. If the
 `ENV_VAR` environment variable is set then the placeholder is replaced to its
-value. Otherwise an error message is logged saying that the variable is not set,
-and in this case an empty string is inserted in the place of the placeholder.
+value. Otherwise an error message is logged saying that the variable is not
+set, and in this case an empty string is inserted in the place of the
+placeholder.
 
-##### Clang Static Analyzer
+##### _Clang Static Analyzer_
 
 Use the `--saargs` argument to a file which contains compilation options.
 
+~~~~
+CodeChecker analyze mylogfile.json --saargs extra_compile_flags.txt -n myProject
+~~~~
+
+Where the `extra_compile_flags.txt` file contains additional compilation
+options, for example:
 
-    CodeChecker check --saargs extra_compile_flags -n myProject -b "make -j4"
+~~~~~
+-I~/include/for/analysis -I$(MY_LIB)/include -DDEBUG
```

Comment:
I'm not sure what this part is about. I just aligned the documentation to have a better wording, but am using the same examples from the previous version.